### DISCUSSION TITLE
Looking for a list of all utility classes.

### DISCUSSION THREAD

**Discussion by rschoonheim** [tailwindlabs/tailwindcss#9368] [Help]
Hello everyone, 

I'm searching a list that contains all utility classes from tailwind. I'm unsure how I can generate, creating it manually is very time consuming :). 

Thanks in advance!

**Comment by wongjn** [comment]
The Tailwind classes can be found in [the documentation](https://tailwindcss.com/docs), grouped into sections. You can also use the site search to find a particular utility class.

**Reply by rschoonheim** [reply]
Thanks Wong, I know the documentation. I need a list of all CSS classes that tailwind introduces. Just the raw definition like:

```
.class1
.class2
.class3
```

Working on a tailwind formatter..

**Reply by wongjn** [reply]
There would be an infinite amount of classes, due to arbitrary values and arbitrary variants which would be impossible to generate. Even without these, if you really wanted every single class possible, you would need to generate every possible combination of the variants, (like `dark:first:hover:focus:md:visited:last:only:…:m-1`) so there would be all the "base" classes plus all these permutations of variants and thus the total number of possible classes would be something like (54![^1] + 1) × number of "base" classes (maths probably not correct here, but something like this). The "real" number would be slightly less since some variants don't make sense together, i.e. `sm:md:lg:xl:m-1` but this obscenely large number should demonstrate the point.

Otherwise, if you are only looking for the "base" classes from Tailwind, you could look at emulating this test which seems to generate all base classes: https://github.com/tailwindlabs/tailwindcss/blob/4fddd2d611f1255eb0371589031ed95b5bab8e68/tests/getClassList.test.js#L4-L28

[^1]: This is 54 factorial, equal to 230,843,697,339,241,380,472,092,742,683,027,581,083,278,564,571,807,941,132,288,000,000,000,00

**Reply by rschoonheim** [reply]
Thanks @wongjn used that test case to generate the "base" classes. Generating `230,843,697,339,241,380,472,092,742,683,027,581,083,278,564,571,807,941,132,288,000,000,000,000` classes is impossible :grimacing:
